typings/react.d.ts

```diff
@@ -0,0 +1,10 @@
+import React from "react"
+
+declare module "react" {
+  /**
+   * This typing only exists in the React 16 typings, but the storybook-react typings assume we’re already on that even
+   * though React 16 is not a hard dependency of storybook-react. Rather than updating to React 16, we simply augment
+   * the React 15 typings to add the missing type.
+   */
+  type ComponentType<P = {}> = React.ComponentClass<P> | React.StatelessComponent<P>;
+}
```

Comment:
Interestingly enough there actually _is_ a React 15 version of the typings that has this type, but when I try to install that all the other typings packages related to react install their own copy of the react typings, which breaks everything https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b13b82c214ac5ed2397ec5f3ab6a157a9fc7e220/types/react/v15/index.d.ts#L78.